evolver/logutils.py

```diff
@@ -6,24 +7,63 @@
 logging.addLevelName(EVENT, "EVENT")
 
 
+class LogInfo:
+    """Helper class to provide "extra" to log records integrated with history.
+
+    This class acts similar to a dict, but with all fields within a dict under a
+    specific key and ensures the fields are json-serializable. Used with the
+    history integration, fields provided will be available in the stored record.
+    This is useful because all fields within extra are placed at the top level
+    of a logrecord dict, so otherwise there is no way to distinguish them.
+
+    Example:
+        self.logger.log(EVENT, "Event message", extra=LogInfo(vial=1))
+        self.logger.info(".", extra={"not_stored": 1, **LogInfo(vial=1).dump()})
+    """
+
+    EXTRA_KEY = "evolver_extra"
+
+    @staticmethod
+    def json_check(data, silent=False):
+        if data == {}:
+            return data
+        try:
+            json.dumps(data)
+            return data
+        except TypeError as e:
+            if silent:
```

Comment:
curious why silent is needed?

Reply:
In order to share this with the history hookup, at that time we do not want to raise an exception, but rather silently drop the data. Users of `LogInfo` ought to always have good data, other users would risk dropping non-jsonable data.